### PR TITLE
Cleanup FileSystemProvider from runtime checks

### PR DESCRIPTION
**Pull request by iSazonov**
## PR Summary

Address second part of #2349

- Replace runtime checks with compile time checks.
- Remove unneeded methods.

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` to the beginning of the title and remove the prefix when the PR is ready.
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] User-facing [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed - Issue link:
- **Testing - New and feature**
    - [ ] Not Applicable or can only be tested interactively
    - **OR**
    - [x] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
        - [ ] [Add `[feature]` if the change is significant or affects feature tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr)

### CODE REVIEW

**Comment by iSazonov** [comment]
:-) I wonder that my first all-in-one commit failed and I had to split on steps.

File: src/System.Management.Automation/namespaces/FileSystemProvider.cs [hunk 1]
```diff
@@ -7618,25 +7618,23 @@ private struct REPARSE_GUID_DATA_BUFFER
...
-                        if (linkTarget != null)
-                            return (new string[] { linkTarget });
+                    if (linkTarget != null)
+                    {
+                        return (new string[] { linkTarget });
                     }
                 }
-
+#endif
                 return InternalGetTarget(fileSysInfo.FullName);
```
Comment (rjmholt) [inline src/System.Management.Automation/namespaces/FileSystemProvider.cs]: I see the definition for this method is gone but a call remains. Is there another definition somewhere?

**Review by rjmholt — APPROVED** [review]
This looks good to me -- much nicer!

File: src/System.Management.Automation/namespaces/FileSystemProvider.cs (hunk 1 above)
Comment (iSazonov) [reply src/System.Management.Automation/namespaces/FileSystemProvider.cs]: Yes, one - wih handle parameter, second with string parameter. First one was removed.

Comment (rjmholt) [reply src/System.Management.Automation/namespaces/FileSystemProvider.cs]: Just checking :smile:

**Review by daxian-dbw — APPROVED** [review]

**Pull request merged**